vtex/loaders/product/extend.ts

```diff
@@ -1,21 +1,73 @@
-import { Product, ProductLeaf } from "../../../commerce/types.ts";
+import { Brand, Product, ProductLeaf } from "../../../commerce/types.ts";
 import { AppContext } from "../../mod.ts";
 import { batch } from "../../utils/batch.ts";
 import { extension as simulateExt } from "../../utils/extensions/simulation.ts";
 import { withIsSimilarTo } from "../../utils/similars.ts";
 import { toReview } from "../../utils/transform.ts";
 import listLoader from "../legacy/productList.ts";
+import brandsLoader from "../legacy/brands.ts";
 
 export interface Props {
   simulate?: boolean;
   similars?: boolean;
   kitItems?: boolean;
   variants?: boolean;
   reviews?: boolean;
+  /**
+   * @description Adds brand information to the product, useful for the Intelligent Search loaders.
+   */
+  brands?: boolean;
 
   products: Product[];
 }
 
+const CACHE_NAME = "vtex-brands";
+const ONE_HOUR = 60 * 60 * 1000;
+
+const brandsExt = async (
+  products: Product[],
+  req: Request,
+  ctx: AppContext,
+) => {
+  const url = new URL(req.url);
+  const brandKey = new URL("/brands", url.origin);
+
+  const cache = await caches.open(CACHE_NAME);
+  const brandsResponse = await cache.match(brandKey);
```

Comment:
avoid `caches` usage

loaders should be cacheable using the cacheKey loader property.

Since this is an extension, you can make a temporary loader that will invoke the brandLoader extension so it will able to cache it (workaround). Or, we need to fix the cacheability of extensions.